### DISCUSSION TITLE
Opening a PR from a commit message

### DISCUSSION THREAD

**Discussion by jhackett-ap** [community/community#58619] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hi folks,

Is there a way to open a PR from a commit message, rather than using the UI or the API? Maybe something like:

```
my commit message here

github:open-pr:<target branch>
```

**Comment by the-officialjosh** [comment]
Unfortunately, there is no direct way to open a pull request (PR) from a commit message in GitHub using Markdown. The process of creating a PR typically requires using the GitHub user interface (UI) or the API.

To create a PR, you need to follow these steps:

1. Push your changes to a branch in the repository.
2. Navigate to the repository on GitHub.
3. Click on the **Pull requests** tab.
4. Click on the **New pull request** button.
5. Select the appropriate branches for the base and compare.
6. Review and provide details for the PR, such as a title and description.
7. Submit the PR for review and merge.

Using the UI or the API ensures that you have the necessary information, including branch details, PR settings, and proper context for reviewers to understand your changes.

While it's not possible to directly create a PR from a commit message in Markdown, you can include references to issues or other PRs in your commit message using specific keywords. For example, including "`Fixes #issue-number`" in a commit message will automatically close the referenced issue upon merging the PR. However, this does not directly create a PR.

If you prefer a more automated approach, you can explore GitHub's API and third-party integrations to create PRs programmatically, but it would still involve interacting with the API rather than directly creating a PR from a commit message.

**Comment by jhackett-ap** [comment]
Ah, that's a shame -could this be marked as a feature request? 

It'd be a good one, and follow on from the logic of being able to undertake an action. Something like `PR <target branch>`.

**Comment by airtower-luna** [comment]
You could make your own automation using Actions, something like this:

* A workflow triggered by `push` events, limited to relevant branches, or maybe simply not the default branch
* Scan the commits in the push (if you use the event payload for this, mind that it [includes only up to 20 commits](https://docs.github.com/en/webhooks-and-events/webhooks/webhook-events-and-payloads#push))
* If you find the code word you want, create a PR for that branch (e.g. using `gh pr create`)

That's not going to work from forks (at least not without the for owner providing a PAT), but if you just want to automate work within your repo it should be fine.

**Reply by the-officialjosh** [reply]
Let me try this
